guides/assets/javascripts/guides.js

```diff
@@ -102,5 +103,91 @@
       }, 3000);
       e.clearSelection();
     });
+
+    var mainColElems = Array.from(document.getElementById("mainCol").children);
+    var navLinks = document.querySelectorAll("#subCol a");
+    var DESKTOP_THRESHOLD = 1024;
+
+    var matchingNavLink = function (elem) {
+      if(!elem) return;
+      var index = mainColElems.indexOf(elem);
+
+      var match;
+      while (index >= 0 && !match) {
+        var link = mainColElems[index].querySelector("a");
+        if (link) {
+          match = document.querySelector('#subCol [href="' + link.getAttribute("href") + '"]');
+        }
+        index--;
+      }
+      return match;
+    }
+
+    var removeHighlight = function () {
+      for (var i = 0, n = navLinks.length; i < n; i++) {
+        navLinks[i].classList.remove("active");
+      }
+    }
+
+    var updateHighlight = function (elem) {
+      removeHighlight();
+      if (!elem) return;
+      elem.classList.add("active");
+      if (window.innerWidth > DESKTOP_THRESHOLD) {
+        elem.scrollIntoView({ block: 'center', inline: 'end' });
+      }
+    }
+
+    var resetNavPosition = function () {
+      var chapters = document.querySelector(".chapters");
+      chapters?.scroll({ top: 0 });
+    }
+
+    var belowBottomHalf = function (i) {
+      return i.boundingClientRect.bottom > (i.rootBounds.bottom + i.rootBounds.top) / 2;
+    }
+
+    var prevElem = function (elem) {
+      var index = mainColElems.indexOf(elem);
+      if (index <= 0) {
+        return null;
+      }
+      return mainColElems[index - 1];
+    }
+
+    var PAGE_LOAD_BUFFER = 1000;
+
+    var navHighlight = function (entries) {
+      entries.forEach(function (entry) {
+        if (entry.time < PAGE_LOAD_BUFFER && !entry.isIntersecting) {
+          return;
+        }
+        else if (belowBottomHalf(entry) && !entry.isIntersecting) {
+          updateHighlight(matchingNavLink(prevElem(entry.target)));
+        } else if (entry.isIntersecting) {
+          updateHighlight(matchingNavLink(entry.target));
+        }
+      });
+    }
+
+    var threshold = window.innerHeight - 50;
```

Comment:
Maybe call this `rootMarginBottom` (it's bottom?), to not get confused with the `threshold` we pass `0` below.